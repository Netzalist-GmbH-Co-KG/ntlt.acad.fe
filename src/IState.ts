export interface ISession {
    currentUser: string,
    sessionStart: Date,
}

export interface IState {
    currentSession: ISession | null,
    applicationStart: string,
    reducerCounter: number,
}
