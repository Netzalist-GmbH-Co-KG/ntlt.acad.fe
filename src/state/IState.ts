export const initialState: IState = {
    demo: { 
        applicationStart: Date(),
        reducerCounter: 0
    },
    currentSession: null
}

export interface ISession {
    currentUser: string,
    sessionStart: Date,
}

export interface IDemo {
    applicationStart: string,
    reducerCounter: number,
}

export interface IState {
    currentSession: ISession | null,
    demo: IDemo
}
