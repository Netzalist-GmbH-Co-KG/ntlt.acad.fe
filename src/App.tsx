import * as React from 'react';
import { connect } from 'react-redux'
import * as Redux from 'redux';
import * as actions from "./actions";
import './App.css';

import DemoComponent from "./components/DemoComponent";
import LogonComponent from "./components/LogonComponent";
import SessionInfoComponent from "./components/SessionInfoComponent";
import logo from './logo.svg';

import { IState } from './state/IState'

export interface IOwnProps {
  version: string
}

// tslint:disable-next-line:no-empty-interface
interface IStateProps extends IState {
}

interface IDispatchProps {
  increaseCounter: () => void
}

type Props = IStateProps & IDispatchProps & IOwnProps

// tslint:disable-next-line:no-empty-interface
interface IInternalState {
}

class App extends React.Component<Props, IInternalState> {

  constructor(props: Props) {
    super(props);
  }

  public render() {
    return (<div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1 className="App-title">Academy Builder</h1>
      </header>
      <SessionInfoComponent/>
      { this.props.currentSession ? <DemoComponent /> : <LogonComponent /> }
    </div>)
  }
}

function mapStateToProps(state: IState, ownProps: IOwnProps): IStateProps {
  return {
    ...state
  }
}

const mapDispatchToProps = (dispatch: Redux.Dispatch): IDispatchProps => 
  Redux.bindActionCreators( 
    { 
      increaseCounter: actions.increaseCounter
    }, dispatch)


export default connect<IStateProps, IDispatchProps, IOwnProps>
  (mapStateToProps, mapDispatchToProps)(App)
