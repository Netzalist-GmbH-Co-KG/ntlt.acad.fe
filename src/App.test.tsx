import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { create as createTestRenderer } from 'react-test-renderer';
import { createStore } from 'redux';
import App from './App';
import { IState } from './IState'

let store:any = null;

beforeEach( () =>{
  store = createStore<IState, any, any, any>((state = { applicationStart: "2014-01-01", reducerCounter: 0, currentSession: null }, action) => state);
});


it('renders without crashing (Smoke Test)', () => {
  const div = document.createElement('div');
  
  ReactDOM.render(<Provider store={store}><App version="test" /></Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders like the last time (snapshots)', () => {

  const component = createTestRenderer(
    <Provider store={store}><App version="test3" /></Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});