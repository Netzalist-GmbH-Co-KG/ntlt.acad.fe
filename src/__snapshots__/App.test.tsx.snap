// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders like the last time (snapshots) 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src={undefined}
    />
    <h1
      className="App-title"
    >
      Academy Builder
    </h1>
  </header>
  <div
    style={
      Object {
        "backgroundColor": "#222",
        "padding": "10px",
        "textAlign": "right",
      }
    }
  >
    <div
      className="badge badge-danger"
    >
      Not logged in
    </div>
     
  </div>
  <div
    className="form-signin"
  >
    <h1
      className="h3 mb-3 font-weight-normal"
    >
      Login
    </h1>
    <label
      className="sr-only"
      htmlFor="inputEmail"
    >
      Email address
    </label>
    <input
      autoFocus={true}
      className="form-control"
      id="username"
      onChange={[Function]}
      placeholder="Email address"
      required={true}
      type="text"
    />
    <label
      className="sr-only"
      htmlFor="inputPassword"
    >
      Password
    </label>
    <input
      className="form-control"
      id="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
    />
    <button
      className="btn btn-lg btn-primary btn-block"
      id="logonButton"
      onClick={[Function]}
    >
      Sign in
    </button>
    <label
      className="badge badge-danger"
    />
  </div>
  <button
    className="btn btn-link"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "9px",
      }
    }
  >
    Show Logs
  </button>
</div>
`;
